cg/cli/demultiplex/copy_novaseqx_data.py

```diff
@@ -0,0 +1,76 @@
+from pathlib import Path
+import os
+import shutil
+from typing import List, Optional
+
+from cg.constants.demultiplexing import DemultiplexingDirsAndFiles
+
+
+def is_copied(analysis_directory: Path) -> bool:
+    return Path(analysis_directory, DemultiplexingDirsAndFiles.COPY_COMPLETE).exists()
+
+
+def is_analyzed(analysis_directory: Path) -> bool:
+    return Path(
+        analysis_directory,
+        DemultiplexingDirsAndFiles.DATA,
+        DemultiplexingDirsAndFiles.ANALYSIS_COMPLETED,
+    ).exists()
+
+
+def is_in_demultiplexed_runs(flow_cell_name: str, demultiplexed_runs: Path) -> bool:
+    return Path(demultiplexed_runs, flow_cell_name).exists()
+
+
+def get_latest_analysis_directory(flow_cell: Path) -> Optional[Path]:
+    analysis_path = Path(flow_cell, DemultiplexingDirsAndFiles.ANALYSIS)
```

Comment:
```suggestion
def get_latest_analysis_directory(flow_cell_dir: Path) -> Optional[Path]:
    analysis_path = Path(flow_cell_dir, DemultiplexingDirsAndFiles.ANALYSIS)
```

Reply:
I agree on this, we use `flow_cell` when we initialise a `FlowCellDirectoryData` or `Flowcell` statusDB model. We should be explicit when we refer to paths/directories as being such.

Reply:
I strongly disagree with you on this point with the reasoning I gave above in mind. It is not about whether the context indicates it is a path but rather that `flow_cell` is consistently used to intialise certain models in the code base. While reviewing the code it was difficult and counterintuitive to see `flow_cell` variable pointing to a path rather than a model of `FlowCellDirectoryData` which we use everywhere in conjunction with flow cells.

In other parts of the codebase we also use a suffix of `path` or `dir` when we refer to directories or files etc. so not doing it here would take away from consistency in our code base, which we are trying to improve.

Reply:
@henrikstranneheim, @diitaz93  what is your opinion about this? Is it enough to infer that the `flow_cell` refers to a directory from context or should we be more explicit?

Reply:
I agree that we should not think about the entire code base, however we  here we are still in the scope of working with flow cells. 

What I am trying to make clear above: I have been working with this the code surrounding flow cells for a while now and I found it confusing to see that `flow_cell` was being used to refer to a path here. So to me not being explicit here  actually increased cognitive load to understand the code, hence the suggested changes.

But I think we might not be able to agree on this, so it is good to get other opinions. Although @diitaz93 suggested the same change.

Reply:
I tend to like being more explicit in these cases to avoid confusion what this particular variable in this context holds.I agree that type hints help a lot, but adding `_path`, `_dir` or `_file` will make it near impossible to misinterpret the contract and also definitely show that we are not expecting the object and we do not even have to read the function code to understand how to use it.

Reply:
Renamed parameters to `flow_cell_dir: Path` 👍